sys-boot/refind/refind-0.11.4.ebuild

```diff
@@ -0,0 +1,197 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs flag-o-matic multiprocessing
+
+DESCRIPTION="The UEFI Boot Manager by Rod Smith"
+HOMEPAGE="https://www.rodsbooks.com/refind/"
+
+SRC_URI="mirror://sourceforge/project/${PN}/${PV}/${PN}-src-${PV}.tar.gz"
+
+LICENSE="BSD GPL-2 GPL-3 FDL-1.3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+FS_USE="btrfs +ext2 +ext4 hfs +iso9660 ntfs reiserfs"
+IUSE="${FS_USE} doc custom-cflags"
+
+DEPEND="|| ( >=sys-boot/gnu-efi-3.0.2 >=sys-boot/udk-2018-r1 )"
+
+DOCS="NEWS.txt README.txt docs/refind docs/Styles"
+
+UDK_ROOT="${EPREFIX}"/usr/share/udk
+[[ -d /usr/share/udk ]] && UDK_ROOT=/usr/share/udk
+UDK_WORKSPACE="${WORKDIR}/udk"
+
+pkg_pretend() {
+	if use custom-cflags; then
+		ewarn
+		ewarn "You have enabled building with USE=custom-cflags. Be aware that"
+		ewarn "using this can result in EFI binaries that fail to run and may"
+		ewarn "fail to build at all. This is strongly advised against by upstream."
+		ewarn
+		ewarn "See https://bugs.gentoo.org/598587#c3 for more information"
+		ewarn
+	fi
+}
+
+pkg_setup() {
+	if use x86; then
+		export EFIARCH=ia32
+		export BUILDARCH=ia32
+	elif use amd64; then
+		export EFIARCH=x64
+		export BUILDARCH=x86_64
+	else
+		# Try to support anyway
+		export BUILDARCH=$( uname -m | sed s,i[3456789]86,ia32, )
+		if [[ ${BUILDARCH} == "x86_64" ]] ; then
+			export EFIARCH=x64
+		else
+			export EFIARCH=${ARCH}
+		fi
+	fi
+}
+
+src_prepare() {
+	default
+
+	# bug 598647 - PIE not supported
+	sed -e 's:CFLAGS          =:& -fno-PIE:' -i Make.common || die
+
+	# Update Makefile
+	sed -e "s:\(EDK2BASE=\).*$:\1${UDK_ROOT}:" \
```

Comment:
Since it's a Makefile, substitute a make variable for the path, and pass the variable to make. This will prevent it from exploding when path accidentally matches your sed chars.

Also, this is really asking for a patch file instead. Possibly replace both seds with a patch.

Reply:
Depending on what you answer above, UDK_ROOT may not be a static value, and so could not be inside a patch. Or I would need to `sed` the patched file…

Reply:
So `UDK_ROOT` depends on `EPREFIX`, then patch is not a good solution here.

Reply:
OK for make variable.

Reply:
In fact, can't you just pass EDK2BASE to emake?

Reply:
I have not tested, but I'm not sure it would work. Actually, EDK2BASE is exported in the Makefile, probably because it is needed in some calls, and simply passing it as a variable to Makefile could not fit the requirements.

Reply:
After testing, this work.